quinn-proto/src/connection/mod.rs

```diff
@@ -490,6 +500,13 @@ impl Connection {
         &self.paths.get(&path_id).expect("known path").data
     }
 
+    /// Gets the [`PathStatus`] for a known [`PathId`].
+    ///
+    /// Will panic if the path_id does not reference any known path.
```

Comment:
FWIW, and I know nothing is final so this is fine, I don't think we can panic here because the application has no way to ensure a PathId they pass in is still valid.

Reply:
hey, this is not Digs doing. He is just passing along the warning I did for getting the path data. This is based on our discussions about `unwrap_or_default` vs `unwrap`. Getting the path data currently has that warning exactly because we want to get screamed at for when we expect a path to exist and it isn't, while developing, so that we can alter this behaviour appropriately.

This is the long version of: Yes, this is not final. Yes, this is intentional